ludwig/api.py

```diff
@@ -523,7 +523,10 @@ def on_epoch_end(self, trainer, progress_tracker, save_path):
                         logger.info(tabulate(experiment_description, tablefmt="fancy_grid"))
 
                         print_boxed("LUDWIG CONFIG")
-                        logger.info(pformat(self.config_obj.to_dict(), indent=4))
+                        logger.info(
+                            "Full config saved to: "
+                            f"{output_directory}/{experiment_name}/model/model_hyperparameters.json"
+                        )
```

Comment:
Very welcome change :)

Reply:
This is a good idea, however, I believe we only have the full config object at this point in the code. I'll add a TODO for introducing the notion of explicitly set fields in the model config object so that we can reconstruct the "user's config".

Reply:
Looks like we are tracking the user config after all! I added a small change to the ModelConfig object to include the user config as an additional attribute.

Reply:
This has been landed as part of #2820